### ISSUE TITLE
Passport::ignoreRoutes() doesn't work

### ISSUE THREAD

**Issue by FrancisMawn** [laravel/passport#1585]
- Passport Version: 11.2.1
- Laravel Version: 9.35.1
- PHP Version: 8.1.11
- Database Driver & Version: MySQL 8.0

### Description:
Hi,

I tried adding `Passport::ignoreRoutes()` to the boot method of my RouteServiceProvider class and it gets called after Passport registered its own routes.

How was that meant to be used? I'm looking for a way to override the routes automatically generated by the package. In #1575 it was mentioned that the routes could be published, but I don't see any options using the `artisan vendor:publish` command...?

**Comment by FrancisMawn** [comment]
Here is what I ended up doing in case anyone come across a similar issue...

Update your **composer.json** file to prevent package auto-discovery for Laravel Passport:
```json
"extra": {
    "laravel": {
        "dont-discover": [
            "laravel/passport"
        ]
    }
}
```

Then update your **config/app.php** file to add the PassportServiceProvider class after your other application service providers:

```php
/*
 * Application Service Providers...
 */
App\Providers\AppServiceProvider::class,
App\Providers\AuthServiceProvider::class,
App\Providers\EventServiceProvider::class,
App\Providers\RouteServiceProvider::class,

// Register Laravel Passport after the RouteServiceProvider class.
Laravel\Passport\PassportServiceProvider::class,
```

Then in your RouteServiceProvider boot method, you can add the `Passport::ignoreRoutes()` method call. Now I can define my own routes file and register it in the Http Kernel and customize them to my liking.

That did the trick, not sure if there was an easier/proper way of doing it.

**Comment by hassanhmd23** [comment]
Hey @FrancisMawn, in order to ignore existing passport routes, you just add `Passport::ignoreRoutes()` to the Register method in the AppServiceProvider. That's how I got it to work.

**Comment by gorczakw** [comment]
The register method is called for each provider before the boot method. You can disable default routing and put your own routing in the boot method e.g. in the RouteServiceProvider class.

**Comment by FrancisMawn** [comment]
@gorczakw Please read my previous comment.

I initially tried to put the `Passport::ignoreRoutes()` declaration into the RouteServiceProvider register method and it broke the whole thing.

For some obscure reasons, putting it into the AppServiceProvider register method seems to work just fine though. Again, Laravel specifies that the register method should not be used for other purposes than binding stuff to the IoC:
>As mentioned previously, within the register method, you should only bind things into the [service container](https://laravel.com/docs/9.x/container). You should never attempt to register any event listeners, routes, or any other piece of functionality within the register method.
 
So although it works, it goes against the framework recommendations.

**Comment by axlon** [comment]
@FrancisMawn you can't call this method in the `boot()` method of your service provider, unless you can somehow make sure that your service provider's `boot()` method is called before Passport's `boot()`.

Did you add `parent::register()` to the route service provider when you added it to register there? Because the `RouteServiceProvider` Laravel ships with actually extends another "route service provider", which already has a `register()` function defined. If you didn't tell PHP to also call that register method then that would be why none of your routes were working.

**Comment by FrancisMawn** [comment]
@axlon You're right! I did not call the `parent::register()` method on the RouteServiceProvider class so that's why none of the routes would register. Thank you for that clarification, I totally missed that :)

For now, I used the boot method with the approach I documented above and everything works. I guess we can close this issue.

I believe the Passport documentation might be missing instructions around proper ways of modifying the package routes. The removal of the static routes method between version 10 and 11 is what lead me here in search of some alternatives.
